ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -460,21 +472,16 @@ private TransactionProcessingResult processTransaction(final Transaction transac
             .createBlockHashLookup(blockchain, blockSelectionContext.pendingBlockHeader());
     final TransactionAccessList transactionAccessList =
         new TransactionAccessList(currentTxnLocation.get());
-    final TransactionProcessingResult result =
-        transactionProcessor.processTransaction(
-            txWorldStateUpdater,
-            blockSelectionContext.pendingBlockHeader(),
-            transaction,
-            blockSelectionContext.miningBeneficiary(),
-            operationTracer,
-            blockHashLookup,
-            TransactionValidationParams.mining(),
-            blockSelectionContext.blobGasPrice(),
-            Optional.of(transactionAccessList));
-    if (txWorldStateUpdater instanceof StackedUpdater<?, ?> stackedUpdater) {
-      balBuilder.addTransactionLevelAccessList(transactionAccessList, stackedUpdater);
-    }
```

Comment:
not blocking for this review. why was the `balBuilder.addTransactionLevelAccessList` call moved from here to later to the handleTransactionSelected method?

Reply:
Because that way we are sure to add only the transactions that are selected in the bal and not those that are in timeout. The reason for the bug you had.